control-plane-operator/hostedclusterconfigoperator/controllers/inplaceupgrader/inplaceupgrader.go

```diff
@@ -618,3 +627,52 @@ func inPlaceUpgradeConfigMap(poolName string, namespace string) *corev1.ConfigMa
 		},
 	}
 }
+
+func compressAndEncodePayload(payload []byte) (*bytes.Buffer, error) {
+	out := bytes.NewBuffer([]byte{})
+
+	// We need to base64-encode our gzipped data so we can marshal it in and out
+	// of a string since ConfigMaps and Secrets expect a textual representation.
+	base64Enc := base64.NewEncoder(base64.StdEncoding, out)
+	defer base64Enc.Close()
+
+	gz, err := gzip.NewWriterLevel(base64Enc, gzip.BestCompression)
+	if err != nil {
+		return nil, fmt.Errorf("could not initialize gzip writer: %w", err)
+	}
+
+	defer gz.Close()
+
+	if _, err := io.Copy(gz, bytes.NewBuffer(payload)); err != nil {
+		return nil, fmt.Errorf("could not compress payload: %w", err)
+	}
+
+	if err := gz.Close(); err != nil {
+		return nil, fmt.Errorf("could not close gzipwriter: %w", err)
+	}
+
+	if err := base64Enc.Close(); err != nil {
+		return nil, fmt.Errorf("could not close base64 encoder: %w", err)
+	}
+
+	return out, nil
+}
+
+func decodeAndDecompressPayload(payload []byte) (*bytes.Buffer, error) {
+	out := bytes.NewBuffer([]byte{})
+
+	base64Dec := base64.NewDecoder(base64.StdEncoding, bytes.NewReader(payload))
+
+	gz, err := gzip.NewReader(base64Dec)
+	defer gz.Close()
```

Comment:
The linter is reporting `should check returned error before deferring gz.Close() (SA5001)` here

Reply:
Fixed. My linter settings also uncovered this:

```console
inplaceupgrader.go:674:15: G110: Potential DoS vulnerability via decompression bomb (gosec)
        if _, err := io.Copy(out, gz); err != nil {
```

I've included a fix and will open a PR to implement the same fix in the MCO repo.